arch/x86/include/asm/kasan.h

```diff
@@ -1,8 +1,9 @@
 #ifndef _ASM_X86_KASAN_H
 #define _ASM_X86_KASAN_H
 
-#define KASAN_SHADOW_START	0xffffd90000000000UL
-#define KASAN_SHADOW_END	0xffffe90000000000UL
+#define KASAN_SHADOW_START      (KASAN_SHADOW_OFFSET + (1ULL << 61) \
```

Comment:
This puzzled me for a while.
Please add a comment explaining that kernel addresses do not start at zero, and so we add kernel_address_space_start>>kasan_shadow_scale to the offset that compiler uses.